### ISSUE TITLE
keepd's documented stack effect seems wrong?

### ISSUE THREAD

**Issue by kwshi** [factor/factor#2384]
`keepd`'s documented stack effect is `( ..a x y quot: ( ..a x y -- ..b x ) -- ..b x )`.  To my understanding, this is incorrect.  As an example, running
```factor
"..a" "x1" "y" [ 3drop "..b" "x2" ] keepd
```
results in the stack
```
"..b"
"x1"
"x2"
```
which is inconsistent with the documented stack effect, which would suggest the resulting stack be only `"..b" "x1"`, or `"..b" "x2"`).

## Suggested fix

`keepd` is defined as `2keep drop`.  The stack effect of `2keep` is
```factor
2keep ( ..a x y quot: ( ..a x y -- ..b ) -- ..b x y )
```
which, composing _only the output_ with `drop ( y -- )`, should result in a stack effect of
```factor
keepd ( ..a x y quot: ( ..a x y -- ..b ) -- ..b x )
```
The main difference being that `quot` outputs only `..b` instead of the currently-documented `..b x`.

This effect is consistent with the actual behavior of `keepd`:
```factor
"..a" "x" "y" [ 3drop "..b" ] keepd

--- Data stack:
"..b"
"x"
```

## Side question

By the way, how is this possible?  Isn't this an error that the stack-checker should be able to catch?  Is there a bug in the stack-checker too, or are quotation-effects not checked, or is it just that primitive combinators such as `keepd` get special treatment and are ignored by the stack-checker?

**Comment by mrjbq7** [comment]
You are correct, thanks!

Part of this is that inline words are not checked directly, the stack-checker checks their uses and perhaps none in the standard image violate that stack effect.

**Comment by mrjbq7** [comment]
(Fixed in a9c201b4200771b7aacd178222e782852586e985)
